### ISSUE TITLE
Question: What parameters dictate the effective memory bandwidth from the caches?

### ISSUE THREAD

**Issue by Alexandre425** [riscv-boom/riscv-boom#630]
**Type of issue**: question

Hi, I've been playing with BOOM's parameters through my work on performance modelling, which also involves scaling the hardware depending on application requirements. 

Scaling the arithmetic performance is simple enough, it's just a matter of changing the issue width of the respective unit, and scaling the front-end to match it. However, I haven't figured out how I could scale memory bandwidth (besides adding more memory units), and what degree of freedom I have there. The [micro-architecture schematic](https://github.com/riscv-boom/riscv-boom/raw/master/docs/figures/uarch.png) has byte/cycle figures, but I haven't figured out if they are fixed or parameterizable.

In short, what parameters can I change to increase or decrease my memory bandwidth?

**Comment by erlingfn** [comment]
Not a boom developer, but I have worked on the boom's memory system as part of my work.
The bandwidth to and from memory is governed largely by which level of the memory hierarchy the load operations hit, and the amount of load operations in-flight at a moment. In order to increase bandwidth from the L1 cache, you need to increase the size of the load queue as well as issue width of memory operations. If you need increase bandwidth from the L2, you need to also scale up the amount of MSHRs(Miss status holding registers) of the L1 cache to increase the number of cache-block misses it can handle at a time to match the size of the load queue. Another way to increase the bandwidth is to increase the beatBytes parameter, ie. the number of bytes transferred between caches each cycle.
